src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private static final Random random = new Random();
+    private static final int MAX_NUMBER = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String figureColor = colorSupplier.getRandomColor();
+
+        switch (figureNumber) {
+            case 0:
+                return new Square(figureColor, random.nextInt(MAX_NUMBER) + 1);
+            case 1:
+                return new Rectangle(figureColor, random.nextInt(MAX_NUMBER) + 1,
+                        random.nextInt(MAX_NUMBER) + 1);
+            case 2:
+                return new RightTriangle(figureColor, random.nextInt(MAX_NUMBER) + 1,
+                        random.nextInt(MAX_NUMBER) + 1);
+            case 3:
+                return new Circle(figureColor, random.nextInt(MAX_NUMBER) + 1);
+            case 4:
+                return new IsoscelesTrapezoid(figureColor, random.nextInt(MAX_NUMBER) + 1,
+                        random.nextInt(MAX_NUMBER) + 1, random.nextInt(MAX_NUMBER) + 1);
+            default:
+                return new IsoscelesTrapezoid(figureColor, random.nextInt(MAX_NUMBER) + 1,

```

Comment:
Why in default case Isosceles Trapezoid returned?

Reply:
I think I can return any figure? Isn't it? I read it in Common mistakes (jv-oop-advanced). If you use a switch case construction in your solution, you may just put the last possible option in the default case.